### PR TITLE
Integrate a new ci framework for travis

### PR DESCRIPTION
**Pull request by rthallisey**
**Describe what this PR does and why we need it**:
New ci framework will move away from have a set of bash scripts to run only mediawiki and postgresql. This framework will allow us to test any apb.

Changes proposed in this pull request
 - Pull in the new ci framework

### CODE REVIEW

**Comment by rthallisey** [comment]
Early WIP. Want to see travis run this.  Also, running ```make run``` to compile to go binary probably isn't a best practice.

**Comment by rthallisey** [comment]
[(25) CI Framework](https://trello.com/c/SSqzLdPs/491-25-ci-framework)

**Comment by rthallisey** [comment]
[(25) CI Framework](https://trello.com/c/SSqzLdPs/491-25-ci-framework)

**Comment by jmrodri** [comment]
# PASS

```
[jesusr@speed3 ansible-service-broker{rthallisey-new-ci-framework}]$ make ci                                                                                                                                                                  
Using OpenShift Cluster                                                                                                                                                                                                                       
ACTION: provision                                                                                                                                                                                                                             
Running: oc create -f /tmp/mediawiki123.yaml                                                                                                                                                                                                  
serviceinstance "mediawiki123" created                                                                                                                                                                                                        
                                                                                                                                                                                                                                              
Waiting for /tmp/mediawiki123.yaml to be ready                                                                                                                                                                                                
Provisioning                                                                                                                                                                                                                                  
Provisioning                                                                                                                                                                                                                                  
Provisioning                                                                                                                                                                                                                                  
Provisioning                                                                                                                                                                                                                                  
Provisioning                                                                                                                                                                                                                                  
Provisioning                                                                                                                                                                                                                                  
Provisioning
Provisioning
Provisioning
Provisioning
Provisioning
Provisioning
Provisioning
Provisioning
Provisioning
Provisioning
Provisioning
Provisioning
Provisioning
Provisioning
Provisioning
Provisioning
Provisioning                                                                                                                                                                                                                         [72/3139]
Provisioning
The instance was provisioned successfully
ACTION: provision
Running: oc create -f /tmp/postgresql.yaml
serviceinstance "postgresql" created

Waiting for /tmp/postgresql.yaml to be ready

Provisioning
Provisioning
Provisioning
Provisioning
Provisioning
Provisioning
Provisioning
Provisioning
Provisioning
Provisioning
Provisioning
Provisioning
The instance was provisioned successfully
ACTION: bind
Running: oc create -f /tmp/postgresql-mediawiki123-bind.yaml
servicebinding "mediawiki-postgresql-binding" created

Waiting for templates/postgresql-mediawiki123-bind.yaml to be ready
Injected bind result
Using Instance Name: mediawiki123
Looking for a Deployment with the SAME name used in your ServiceInstance: mediawiki123
deploymentconfig "mediawiki123" updated
ACTION: verify
REPO: rthallisey/service-broker-ci
URL: https://raw.githubusercontent.com/rthallisey/service-broker-ci/master//verify-mediawiki-postgresql.sh
Running: https://raw.githubusercontent.com/rthallisey/service-broker-ci/master//verify-mediawiki-postgresql.sh
Running: curl mediawiki123-ansible-service-broker.172.17.0.1.nip.io/index.php/Main_Page| grep "div class" | cut -f 2 -d "'"
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   621  100   621    0     0    397      0  0:00:01  0:00:01 --:--:--   398
Failed to gather data from mediawiki123-ansible-service-broker.172.17.0.1.nip.io/index.php/Main_Page
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   621  100   621    0     0   3307      0 --:--:-- --:--:-- --:--:--  3320
Failed to gather data from mediawiki123-ansible-service-broker.172.17.0.1.nip.io/index.php/Main_Page
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   621  100   621    0     0   8236      0 --:--:-- --:--:-- --:--:--  8280
Failed to gather data from mediawiki123-ansible-service-broker.172.17.0.1.nip.io/index.php/Main_Page
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 13741    0 13741    0     0  25085      0 --:--:-- --:--:-- --:--:-- 25074
SUCCESS
You can double check by opening http://mediawiki123-ansible-service-broker.172.17.0.1.nip.io/index.php/Main_Page in your browser

ACTION: unbind
Running: oc delete -f /tmp/postgresql-mediawiki123-bind.yaml
servicebinding "mediawiki-postgresql-binding" deleted

ACTION: deprovision
Running: oc delete -f /tmp/mediawiki123.yaml
serviceinstance "mediawiki123" deleted
NAME           AGE
mediawiki123   6m

NAME           AGE
mediawiki123   6m

NAME           AGE
mediawiki123   6m

NAME           AGE
mediawiki123   7m

NAME           AGE
mediawiki123   7m

NAME           AGE
mediawiki123   7m

ACTION: deprovision
Running: oc delete -f /tmp/postgresql.yaml
serviceinstance "postgresql" deleted

NAME         AGE
postgresql   2m

NAME         AGE
postgresql   3m

NAME         AGE
postgresql   3m

NAME         AGE
postgresql   3m

NAME         AGE
postgresql   3m

NAME         AGE
postgresql   3m

NAME         AGE
postgresql   3m

```

**Comment by jmrodri** [comment]
@jmontleon when you get a chance can you review this as well?

**Comment by jmontleon** [comment]
ACK from me. I'd just like to wait til after Wednesday so we can concentrate on any 3.7 issues until we're due and then we can drop this in first thing after, if that's good with everyone else.

File: Makefile [hunk 1]
```diff
@@ -82,7 +82,8 @@ cleanup-ci: ## Cleanup after ci run
 	./scripts/broker-ci/cleanup-ci.sh
 
 ci: ## Run the CI workflow locally
-	./scripts/broker-ci/local-ci.sh
+	@go get github.com/rthallisey/service-broker-ci/cmd/ci
+	@ci
```
[the remark this answers is not in the document]
Comment (rthallisey) [reply Makefile]: ```go get github.com/rthallisey/service-broker-ci/cmd/ci``` will add that library to your $GO_PATH and compile and install the binary, if there is one, using the full git path. That way, you end up with ```~/bin/ci``` installed on the system.

**Comment by jmrodri** [comment]
We have decided per review to wait until Wednesday November 8th, 2017 before merging. At that point it will be ready to go.

**Comment by rthallisey** [comment]
Let's merge the ansibleplaybookbundle CI first so I can reference those templates in config.yaml

depends-on: https://github.com/ansibleplaybookbundle/rhscl-postgresql-apb/pull/7
depends-on: https://github.com/ansibleplaybookbundle/mediawiki123-apb/pull/8

**Comment by jmrodri** [comment]
Ok both dependent PRs were merged.

https://github.com/ansibleplaybookbundle/mediawiki123-apb/pull/8
https://github.com/ansibleplaybookbundle/rhscl-postgresql-apb/pull/7

Merging this now.

**Pull request merged**